crates/host/src/wasmbus/mod.rs

```diff
@@ -2481,14 +2480,23 @@ impl Host {
                 self.start_actor(entry, actor, actor_ref, count, host_id, annotations)
                     .await?;
             }
-            (hash_map::Entry::Occupied(entry), None) => {
-                self.stop_actor(entry, host_id).await?;
-            }
-            (hash_map::Entry::Occupied(entry), Some(count)) => {
+            (hash_map::Entry::Occupied(entry), _) => {
                 let actor = entry.get();
-                let mut instances = actor.instances.write().await;
+                let mut actor_instances = actor.instances.write().await;
                 let count = usize::from(count);
-                let current = instances.values().map(Vec::len).sum();
+                // Only consider instances that match the requested annotations
+                let current_instances = actor_instances
```

Comment:
Naming supernit: something like `matching_instances` might fit better. `current` kinda makes it sound like it's all "running" instances